src/components/main/files/mod.rs

```diff
@@ -31,10 +35,59 @@ pub fn Files(cx: Scope<Props>) -> Element {
     let show_new_folder = use_state(&cx, || false);
     let show_upload = use_state(&cx, || false);
 
+    let file_storage = cx.props.storage.clone();
+
+    let root_directory = match file_storage.current_directory() {
+        Ok(current_directory) => current_directory, 
+        Err(error) => {
+            log::error!("Not possible to get root directory, error: {:?}", error);
+            Directory::default()
+        },
+    };   
+
+    let parent_directory = use_ref(&cx, || root_directory.clone());
+    let parent_dir_items = use_ref(&cx,  HashSet::new);
+
+
+    
+    use_future(&cx, (&file_storage, parent_directory, parent_dir_items), 
+    |(mut file_storage, parent_directory, parent_dir_items)| 
+    async move {
+      let parent_dir = parent_directory.with(|dir| dir.clone());
+      if parent_dir.name() == "root" {
+        loop {
+            match file_storage.root_directory().get_item("main_directory") {
+                Ok(item) => {
+                    match item.get_directory() {
+                        Ok(directory) => {
+                            parent_directory.with_mut(|dir| *dir = directory.clone());
+                            parent_dir_items.with_mut(|_| directory.get_items());
```

Comment:
I know that in some areas using `write` or `read` may not be preferred (since if youre holding them to long it can cause problems), but usually in this case writing to the reference using the code i given wont be a problem since the write is dropped (or internally its mutable borrow from the underlining reference) but it can be left as is.